### PR TITLE
Adds video and image list fields to VisionRequest

### PR DESCRIPTION
**Pull request by maciekrb**
Adds proposal for video and image list fields that can be used for vision requests. The goal is to make `VisionRequest` a general purpose message, that we can use across different systems, the same way we are currently using `VisionResult`  messages.

### CODE REVIEW

File: vision/vision_service.proto [hunk 1]
```diff
@@ -19,6 +21,14 @@ message VisionRequest {
   // Recognitions to perform on the image. The recognitions should be done
   // after the detections.
   repeated EnumFacialRecognitionTag recognition = 3;
+
+  // It is sometimes more convenient to send many images to be processed
```
Comment (arhuaco) [inline vision/vision_service.proto]: // s/many images/more than one image/

File: vision/vision_service.proto [hunk 2]
```diff
@@ -19,6 +21,14 @@ message VisionRequest {
...
+  // by vision services
+  ImageList image_list = 4;
+
+  // or video data if splitting into frames is not convenient
+  // in the client.
```
Comment (arhuaco) [inline vision/vision_service.proto]: There is a possibility: That you get 1000 images at 10fps an then encode to save bandwidth.  It's better not to anticipate how the messages will be used.

File: vision/vision.proto [hunk 3]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
...
 
+// Collection of images used as source for
+// a detection
+message ImageList {
+  // image data 
+  repeated bytes data = 1;
+
+  // if image data above was taken in a sequence, the fps should 
+  // describe how many of such frames were taken in a second
+  int32 fps = 2;
```
Comment (arhuaco) [inline vision/vision.proto]: I would use frames_per_second here. It's better not to use abbreviations. This way you can spare the comments in code.

File: vision/vision.proto [hunk 4]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
   bytes image_small = 7;
 }
 
+// Collection of images used as source for
+// a detection
```
Comment (maciekrb) [inline vision/vision.proto]: make sense, updating

File: vision/vision.proto [hunk 5]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
...
+message ImageList {
+  // image data
```
Comment (maciekrb) [inline vision/vision.proto]: ACK

File: vision/vision.proto [hunk 6]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
   bytes image_small = 7;
 }
 
+// Collection of images used as source for
+// a detection
+message ImageList {
+  // image data 
+  repeated bytes data = 1;
+
+  // if image data above was taken in a sequence, the fps should
```
Comment (maciekrb) [inline vision/vision.proto]: agree, updating

File: vision/vision.proto (hunk 3 above)
Comment (maciekrb) [inline vision/vision.proto]: updating

File: vision/vision.proto [hunk 7]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
...
+enum EnumVideoCodec {
+  UNDEFINED = 0;
+  H264 = 1;
+  MP4V = 2;
+  RV24 = 3;
+  VP8 = 4;
+  VP9 = 5;
+}
+
+// Source video for recognitions.
```
Comment (maciekrb) [inline vision/vision.proto]: makes sense as well

File: vision/vision.proto [hunk 8]
```diff
@@ -161,6 +161,37 @@ message RectangularDetection {
...
+message Video {
+  // RGB video data
+  bytes data = 1;
+
+  // Video codec used to encode data
+  EnumVideoCodec codec = 2;
+}
+
 // This proto organizes (at least) four components:
```
Comment (maciekrb) [inline vision/vision.proto]: done

File: vision/vision_service.proto [hunk 9]
```diff
@@ -9,6 +9,8 @@ package admobilize_vision;
 
 import "vision.proto";
 
+// This proto aims at providing a flexible structure for client applications
```
Comment (maciekrb) [inline vision/vision_service.proto]: ok with this as well

File: vision/vision_service.proto (hunk 1 above)
Comment (maciekrb) [inline vision/vision_service.proto]: done

File: vision/vision_service.proto (hunk 2 above)
Comment (maciekrb) [inline vision/vision_service.proto]: good as well

**Pull request merged**